From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Sat, 19 Sep 2026 16:54:28 +0200
Subject: userVerifier: Reuse user verifier proxies on resets

Avoid re-creating the same user verifiers all times we may (soft) reset
and that could happen a lot during a single authentication phase.

So just bind each userVerifier to an user and avoid creating a new one
in case we're requesting a new user verifier for the same user
---
 js/gdm/userVerifier.js | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/js/gdm/userVerifier.js b/js/gdm/userVerifier.js
index 56d4fd0..ba13b9a 100644
--- a/js/gdm/userVerifier.js
+++ b/js/gdm/userVerifier.js
@@ -97,7 +97,7 @@ export class ShellUserVerifier extends Signals.EventEmitter {
 
         try {
             const proxies = await this._getUserVerifierProxies(userName, this._cancellable);
-            this._setUserVerifier(proxies.userVerifier);
+            this._setUserVerifier(userName, proxies.userVerifier);
 
             let lastInitError;
             let anyBegan = false;
@@ -205,9 +205,13 @@ export class ShellUserVerifier extends Signals.EventEmitter {
         this._clearUserVerifier();
     }
 
-    _setUserVerifier(userVerifier) {
+    _setUserVerifier(userName, userVerifier) {
+        if (this._userVerifier?.userName === userName)
+            return;
+
         this._clearUserVerifier();
         this._userVerifier = userVerifier;
+        this._userVerifier.userName = userName;
         this._userVerifier.get_connection().connectObject(
             'closed', () => this._clearUserVerifier(),
             this);
